### PR TITLE
REST Client Reactive quickstart

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
Guide PR: https://github.com/quarkusio/quarkus/pull/16441

**Check list**:
Your pull request:

- [X] targets the `development` branch
- [X] uses the `999-SNAPSHOT` version of Quarkus
- [X] has tests (`mvn clean test`)
- [X] works in native (`mvn clean package -Pnative`)
- [X] has native tests (`mvn clean verify -Pnative`)
- [ ] makes sure the associated guide must not be updated
- [X] links the guide update pull request (if needed)
- [ ] updates or creates the `README.md` file (with build and run instructions)
- [X] for new quickstart, is located in the directory _component-quickstart_
- [X] for new quickstart, is added to the root `pom.xml` and `README.md`

### CODE REVIEW

File: .github/workflows/build-pr-development.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: [ 8, 11 ]
+        java: [ 11 ]
```
Comment (michalszynkiewicz) [inline .github/workflows/build-pr-development.yml:14]: @gsmet is this okay? Quarkus no longer seems to build with 8: https://github.com/quarkusio/quarkus-quickstarts/pull/829/checks?check_run_id=2341866380

**Comment by geoand** [comment]
LGTM apart from the JDK 8 question

File: .github/workflows/build-pr-development.yml (hunk 1 above)
Comment (michalszynkiewicz) [reply .github/workflows/build-pr-development.yml:14]: I see the link redirects now to a newer test run. Building Quarkus failed, thje error message was about the `--release` flag.

Comment (michalszynkiewicz) [reply .github/workflows/build-pr-development.yml:14]: here's the failure for another quickstart I'm adding: https://github.com/quarkusio/quarkus-quickstarts/pull/832/checks?check_run_id=2342157467

**Comment by michalszynkiewicz** [comment]
the current failure is unrelated

**Review by geoand — APPROVED** [review]

File: rest-client-reactive-quickstart/pom.xml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.acme</groupId>
+  <artifactId>rest-client-reactive-quickstart</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <properties>
+    <compiler-plugin.version>3.8.1</compiler-plugin.version>
+    <maven.compiler.parameters>true</maven.compiler.parameters>
+    <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (cescoffier) [inline rest-client-reactive-quickstart/pom.xml:11]: 11 no ?
We are dropping java 8.

File: rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm [hunk 3]
```diff
@@ -0,0 +1,54 @@
+####
```
Comment (cescoffier) [inline rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm:1]: We could drop the docker files if they are not referenced in the guide.

File: rest-client-reactive-quickstart/src/main/java/org/acme/rest/client/Country.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package org.acme.rest.client;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (cescoffier) [inline rest-client-reactive-quickstart/src/main/java/org/acme/rest/client/Country.java:7]: Isn't that the default? @geoand

File: rest-client-reactive-quickstart/src/main/resources/META-INF/resources/index.html [hunk 5]
```diff
@@ -0,0 +1,175 @@
+<!DOCTYPE html>
```
Comment (cescoffier) [inline rest-client-reactive-quickstart/src/main/resources/META-INF/resources/index.html:1]: Should probably be removed too.

It has reference on Quarkus versions.... which would is updated regularly.

File: rest-client-reactive-quickstart/src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,2 @@
+country-api/mp-rest/url=https://restcountries.eu/rest
+country-api/mp-rest/scope=javax.inject.Singleton
```
Comment (cescoffier) [inline rest-client-reactive-quickstart/src/main/resources/application.properties:2]: At some point, we should make that the default no?
I always have this line.

@mkouba @geoand @Ladicek

**Review by cescoffier — DISMISSED** [review]
I've made a few comments, but it looks good to me.

File: rest-client-reactive-quickstart/src/main/java/org/acme/rest/client/Country.java (hunk 4 above)
Comment (geoand) [reply rest-client-reactive-quickstart/src/main/java/org/acme/rest/client/Country.java:7]: Good point! I had completely forgot about this enhancement!

File: rest-client-reactive-quickstart/src/main/resources/application.properties (hunk 6 above)
Comment (geoand) [reply rest-client-reactive-quickstart/src/main/resources/application.properties:2]: I agree, I never understood why it's not... The spec maybe?

Comment (geoand) [reply rest-client-reactive-quickstart/src/main/resources/application.properties:2]: Actually, it's best if it's `@ApplicationScoped` in order to make mocking possible

Comment (Ladicek) [reply rest-client-reactive-quickstart/src/main/resources/application.properties:2]: Per the specification https://download.eclipse.org/microprofile/microprofile-rest-client-2.0/microprofile-rest-client-spec-2.0.html#restcdi, the default scope is `@Dependent`.

File: rest-client-reactive-quickstart/src/main/resources/META-INF/resources/index.html (hunk 5 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/resources/META-INF/resources/index.html:1]: removed

File: rest-client-reactive-quickstart/src/main/resources/application.properties (hunk 6 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/resources/application.properties:2]: I'll drop it, we have `@ApplicationScoped` by default which makes sense.

File: rest-client-reactive-quickstart/pom.xml (hunk 2 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/pom.xml:11]: CI didn't like it when I started, let's see what it says now

File: rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm (hunk 3 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm:1]: how strong do you feel on this Clement? I think they bring some value here, even if they are not mentioned...

Comment (cescoffier) [reply rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm:1]: I don't mind keeping them. It's just a pain when we want to add a new one, or when ubi needs to be updated.

File: rest-client-reactive-quickstart/src/main/resources/application.properties (hunk 6 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/resources/application.properties]: done

File: rest-client-reactive-quickstart/src/main/java/org/acme/rest/client/Country.java (hunk 4 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/java/org/acme/rest/client/Country.java]: done

File: rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm (hunk 3 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/docker/Dockerfile.jvm]: done

File: rest-client-reactive-quickstart/src/main/resources/application.properties (hunk 6 above)
Comment (michalszynkiewicz) [reply rest-client-reactive-quickstart/src/main/resources/application.properties]: @Ladicek rest-client-reactive uses `@ApplicationScoped` by default, see https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/rest-client-reactive.adoc#known-limitations

Comment (Ladicek) [reply rest-client-reactive-quickstart/src/main/resources/application.properties]: Didn't know that, thanks for info.

**Pull request merged**